dev_tools/utils/nxdl_utils.py

```diff
@@ -260,59 +259,35 @@ def belongs_to_capital(params):
 
 def get_local_name_from_xml(element):
     """Helper function to extract the element tag without the namespace."""
-    return element.tag[element.tag.rindex("}") + 1 :]
+    return remove_namespace_from_tag(element.tag)
 
 
 def get_own_nxdl_child_reserved_elements(child, name, nxdl_elem):
     """checking reserved elements, like doc, enumeration"""
-    if get_local_name_from_xml(child) == "doc" and name == "doc":
-        if nxdl_elem.get("nxdlbase"):
-            child.set("nxdlbase", nxdl_elem.get("nxdlbase"))
-            child.set("nxdlbase_class", nxdl_elem.get("nxdlbase_class"))
-            child.set("nxdlpath", nxdl_elem.get("nxdlpath") + "/doc")
-        return child
-    if get_local_name_from_xml(child) == "enumeration" and name == "enumeration":
-        if nxdl_elem.get("nxdlbase"):
-            child.set("nxdlbase", nxdl_elem.get("nxdlbase"))
-            child.set("nxdlbase_class", nxdl_elem.get("nxdlbase_class"))
-            child.set("nxdlpath", nxdl_elem.get("nxdlpath") + "/enumeration")
-        return child
+    local_name = get_local_name_from_xml(child)
+    if local_name == "doc" and name == "doc":
+        return set_nxdlpath(child, nxdl_elem, tag_name=name)
+
+    if local_name == "enumeration" and name == "enumeration":
+        return set_nxdlpath(child, nxdl_elem, tag_name=name)
     return False
 
 
 def get_own_nxdl_child_base_types(child, class_type, nxdl_elem, name, hdf_name):
-    """checking base types of group, field,m attribute"""
+    """checking base types of group, field, attribute"""
     if get_local_name_from_xml(child) == "group":
         if (
             class_type is None or (class_type and get_nx_class(child) == class_type)
         ) and belongs_to(nxdl_elem, child, name, class_type, hdf_name):
-            if nxdl_elem.get("nxdlbase"):
-                child.set("nxdlbase", nxdl_elem.get("nxdlbase"))
-                child.set("nxdlbase_class", nxdl_elem.get("nxdlbase_class"))
-                child.set(
-                    "nxdlpath", nxdl_elem.get("nxdlpath") + "/" + get_node_name(child)
-                )
-            return child
+            return set_nxdlpath(child, nxdl_elem)
```

Comment:
Does the set_nxdlpath function perform the same tasks as lines 290-293 do on the LHS ?